pkg/controller/networkpolicy/controller_querier_replier.go

```diff
@@ -150,6 +153,35 @@ func (eq EndpointQueryReplier) QueryNetworkPolicies(namespace string, podName st
 	}
 	// make rules
 	responseRules := make([]Rule, 0)
+	// create rules based on egress and ingress policies
+	for _, internalPolicy := range egress {
+		newRule := Rule{
+			PolicyRef: PolicyRef{
+				Namespace: internalPolicy.Namespace,
+				Name:      internalPolicy.Name,
+				UID:       internalPolicy.UID,
+			},
+			Direction: networkingv1beta1.DirectionOut,
+			//TODO: I am not sure what these are
```

Comment:
See comment above. K8s NetworkPolicies (https://kubernetes.io/docs/concepts/services-networking/network-policies/) support multiple ingress / egress rules. This is meant to identify which one of these rules is selecting the considered endpoint. I believe this information is preserved in the `Rules` field of the `antreatypes.NetworkPolicy` type (meaning the order is the same so you can just use the position in the `Rules` array as the index).

This code seems to confirm that the information is preserved: https://github.com/vmware-tanzu/antrea/blob/master/pkg/controller/networkpolicy/networkpolicy_controller.go#L523

Reply:
What do the ports represents?

Reply:
Actually I think we can remove ports from our `Rule` struct here. With the `RuleIndex` we can identify which ingress / egress rule includes that endpoint. That narrows it down to this: https://godoc.org/k8s.io/api/networking/v1#NetworkPolicyIngressRule. There is another level of "nesting" since each rule can have more than one peer selector, but let's not worry about that.

So let's remove `Ports` and focus on populating `RuleIndex` correctly as I described above.

@jakesokol1